boards/DIM/Src/main.c

```diff
@@ -510,6 +523,63 @@ static void MX_SPI2_Init(void)
     /* USER CODE END SPI2_Init 2 */
 }
 
+/**
+ * @brief TIM2 Initialization Function
+ * @param None
+ * @retval None
+ */
+static void MX_TIM2_Init(void)
+{
+    /* USER CODE BEGIN TIM2_Init 0 */
+
+    /* USER CODE END TIM2_Init 0 */
+
+    TIM_ClockConfigTypeDef  sClockSourceConfig = { 0 };
+    TIM_MasterConfigTypeDef sMasterConfig      = { 0 };
+
+    /* USER CODE BEGIN TIM2_Init 1 */
+
+    /* USER CODE END TIM2_Init 1 */
+    htim2.Instance         = TIM2;
+    htim2.Init.Prescaler   = TIM2_PRESCALER - 1;
```

Comment:
@markomarkusi Is the `-1` needed or not? I always forget

Reply:
Yep 👍 that's right a value of 0 = prescaler of 1x and a value of 1 - prescaler of 2x

Reply:
What about for period?

Reply:
It's the value that it counts down from right? So if my clock is 1Hz and I set period as 2, it would count down from: 2 - 1 - 0 - 2 - 1 - 0. This seems like the `-1` isn't needed?

Reply:
Oh wait maybe you do need the `-1` based on the way I just counted above

Reply:
Ya because you count through 0 as well